pkg/webhook/mutatingwebhook/vm_mutate.go

```diff
@@ -0,0 +1,87 @@
+package mutatingwebhook
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/codeready-toolchain/member-operator/pkg/webhook/mutatingwebhook/types"
+
+	"github.com/pkg/errors"
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var vmLogger = logf.Log.WithName("virtual_machines_mutating_webhook")
+
+func HandleMutateVirtualMachines(w http.ResponseWriter, r *http.Request) {
+	handleMutate(vmLogger, w, r, vmMutator)
+}
+
+func vmMutator(admReview admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
+
+	// unmarshal the object to be sure that it's a VirtualMachine
+	vm := &types.VirtualMachine{}
+	if err := json.Unmarshal(admReview.Request.Object.Raw, vm); err != nil {
+		vmLogger.Error(err, "unable unmarshal VirtualMachine json object", "AdmissionReview", admReview)
+		return responseWithError(errors.Wrapf(err, "unable unmarshal VirtualMachine json object - raw request object: %v", admReview.Request.Object.Raw))
+	}
+
+	patchType := admissionv1.PatchTypeJSONPatch
+	resp := &admissionv1.AdmissionResponse{
+		Allowed:   true,
+		UID:       admReview.Request.UID,
+		PatchType: &patchType,
+	}
+	resp.AuditAnnotations = map[string]string{
+		"virtual_machines_mutating_webhook": "the resource limits were set",
+	}
+
+	// instead of changing the object we need to tell K8s how to change the object
+	vmPatchItems := []map[string]interface{}{}
+
+	// ensure limits are set
+	vmPatchItems = ensureLimits(vm, vmPatchItems)
+
+	patchContent, err := json.Marshal(vmPatchItems)
+	if err != nil {
+		vmLogger.Error(err, "unable to marshal patch items for VirtualMachine", "AdmissionReview", admReview, "Patch-Items", vmPatchItems)
+		return responseWithError(errors.Wrapf(err, "unable to marshal patch items for VirtualMachine - raw request object: %v", admReview.Request.Object.Raw))
```

Comment:
minor thing - seems like it's missing a test case, not sure if you want to add one so that it's reaches 100% coverage :)

Reply:
It's not straightforward to test since `vmPatchItems` is returned by the `ensureLimits` func and I don't see how it would return something that would cause the marshalling to fail. I suppose that means we probably don't need to check for error here but I'd rather have it there just as a safeguard in case we're missing something.

If we really want to test it I'd have to have a fake ensureLimits func or do some refactoring and that doesn't seem worth it to do just for this purpose. So I'm okay with less than 100% coverage and the user pods mutate tests have the same gap in testing :)

Reply:
Yeah. Such things is pretty tricky to test. IMO that's OK to skip it.
But I like your attitude to have 100% coverage when possible @mfrancisc! :)